src/App.tsx

```diff
@@ -1,25 +1,39 @@
-import React from "react";
-import { Routes, Route } from "react-router-dom";
+import { useEffect } from "react";
+import { Routes, Route, useNavigate, Navigate } from "react-router-dom";
 import NotFoundPage from "./Components/home/components/NotFoundPage";
 import Product from "./Components/home/components/Product";
 import HomePage from "./Components/home/HomePage";
 import LoginPage from "./Components/login/LoginPage";
 import LoginEmailPass from "./Components/update/LoginEmailPass";
 import UpdatePassword from "./Components/update/UpdatePassword";
+import Layout from "./layout/Layout";
 
 function App() {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    (async () => {
+      if (!localStorage.getItem("jwt_token")) {
```

Comment:
```suggestion
      if (!localStorage.getItem(LOCAL_STORAGE_KEYS.JWT_TOKEN)) {
```

export const LOCAL_STORAGE_KEYS = {
  JWT_TOKEN: "jwt_token",
  JWT_REFRESH_TOKEN: "jwt_refresh_token"
}